utils/swift_build_support/tests/test_workspace.py

```diff
@@ -0,0 +1,145 @@
+# tests/test_workspace.py ---------------------------------------*- python -*-
+#
+# This source file is part of the Swift.org open source project
+#
+# Copyright (c) 2014 - 2016 Apple Inc. and the Swift project authors
+# Licensed under Apache License v2.0 with Runtime Library Exception
+#
+# See http://swift.org/LICENSE.txt for license information
+# See http://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+#
+# ----------------------------------------------------------------------------
+
+import argparse
+import itertools
+import os
+import shutil
+import tempfile
+import unittest
+
+from swift_build_support.workspace import (
+    Workspace,
+    compute_build_subdir,
+)
+
+
+class WorkspaceTestCase(unittest.TestCase):
+
+    def test_workspace(self):
+        tmpdir1 = os.path.realpath(tempfile.mkdtemp())
+        tmpdir2 = os.path.realpath(tempfile.mkdtemp())
+        os.makedirs(os.path.join(tmpdir1, 'foo'))
+
+        workspace = Workspace(source_root=tmpdir1,
+                              build_root=tmpdir2)
+
+        self.assertEqual(workspace.source_root, tmpdir1)
+        self.assertEqual(workspace.build_root, tmpdir2)
+
+        # source_dir
+        self.assertEqual(workspace.source_dir('foo'),
+                         os.path.join(tmpdir1, 'foo'))
+
+        # build_dir
+        self.assertEqual(workspace.build_dir('target', 'product'),
+                         os.path.join(tmpdir2, 'product-target'))
+
+        shutil.rmtree(tmpdir1)
+        shutil.rmtree(tmpdir2)
+
+
+class ComputeBuildSubdirTestCase(unittest.TestCase):
+
+    def create_basic_args(self, generator, variant, assertions):
```

Comment:
Two reason.
* To make sure `create_basic_args` uses only these properties. If other properties are used in `create_basic_args`, these test should fail.
* We have not yet modularized `argparse.ArgumentParser` creation into `swift_build_support` package. It's hard to import it from unit test.